src/table/use-sticky-columns-context.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { createContext, useContext, useEffect, useCallback, useRef, ReactNode } from 'react';
+
+interface StickyColumnsContextValue {
+  registerChildCallback: (callback: (entry: any) => void) => void;
+  unregisterChildCallback: (callback: (entry: any) => void) => void;
+}
+
+const StickyColumnsContext = createContext<StickyColumnsContextValue | null>(null);
+
+export const useStickyColumnsContext = () => {
+  const context = useContext(StickyColumnsContext);
+  if (!context) {
+    return { registerChildCallback: () => {}, unregisterChildCallback: () => {} };
+  }
+  const { registerChildCallback, unregisterChildCallback } = context as StickyColumnsContextValue;
+  return { registerChildCallback, unregisterChildCallback };
```

Comment:
You could put `left` and `right` directly into the context, no need to over-optimize and dispatch the events manually


Moreover, your code should perform worse on React < 18, because you did not wrap `childCallbacks.current.forEach` into `batchedUpdates` like all custom dispatching should do

Reply:
Good call about the `batchedUpdates` - will wrap it.

Regarding having left/right living at the provider level, this will cause a re-render on all cells, therefore I'm keeping the current pub/sub pattern - I think it's as efficient as we can get it to be with the current Table architecture.